sdk/src/main/java/com/horizen/account/transaction/EthereumTransaction.java

```diff
@@ -0,0 +1,412 @@
+            @NotNull BigInteger nonce,
+            @NotNull BigInteger gasPrice,
+            @NotNull BigInteger gasLimit,
+            @Nullable BigInteger value,
+            @Nullable String data,
+            @Nullable SignatureData signature
+    ) {
+        this(signature != null ?
+                new SignedRawTransaction(
+                        RawTransaction.createTransaction(
+                                nonce,
+                                gasPrice,
+                                gasLimit,
+                                to != null ? to : "",
+                                value != null ? value :
+                                        BigInteger.ZERO,
+                                data
+                        ).getTransaction(),
+                        signature) :
+                RawTransaction.createTransaction(nonce, gasPrice, gasLimit, to, value, data)
+        );
+    }
+
+    // creates an eip1559 transaction
+    public EthereumTransaction(
+            long chainId,
+            @Nullable String to,
+            @NotNull BigInteger nonce,
+            @NotNull BigInteger gasLimit,
+            @NotNull BigInteger maxPriorityFeePerGas,
+            @NotNull BigInteger maxFeePerGas,
+            @Nullable BigInteger value,
+            @Nullable String data,
+            @Nullable SignatureData signature
+    ) {
+        this(
+                signature != null ?
+                        new SignedRawTransaction(
+                                RawTransaction.createTransaction(chainId, nonce, gasLimit, to != null ? to : "", value != null ? value :
+                                        BigInteger.ZERO, data, maxPriorityFeePerGas, maxFeePerGas).getTransaction(), signature)
+                        : RawTransaction.createTransaction(chainId, nonce, gasLimit, to != null ? to : "", value != null ? value :
+                        BigInteger.ZERO, data, maxPriorityFeePerGas, maxFeePerGas)
+        );
+    }
+
+    public RawTransaction getTransaction() {
+        return this.transaction;
+    }
+
+    public boolean isSigned() {
+        return this.transaction instanceof SignedRawTransaction;
+    }
+
+    @Override
+    public byte transactionTypeId() {
+        return AccountTransactionsIdsEnum.EthereumTransaction.id();
+    }
+
+    @Override
+    @JsonProperty("id")
+    public String id() {
+        byte[] encodedMessage;
+        if (this.isSigned()) {
+            SignedRawTransaction stx = (SignedRawTransaction) this.transaction;
+            encodedMessage = TransactionEncoder.encode(this.getTransaction(),
+                    stx.getSignatureData());
+        } else encodedMessage = TransactionEncoder.encode(this.getTransaction());
+        return BytesUtils.toHexString(Hash.sha3(encodedMessage, 0, encodedMessage.length));
+    }
+
+    @Override
+    public byte version() {
+        if (transaction.getType() == TransactionType.LEGACY)
+            return 0x0;
+        return 0x2;
```

Comment:
Cosmetic: we support only legacy (which is `null`) and eip1559 types. Here it would be probably better to use: 
`return TransactionType.EIP1559.getRlpType().byteValue()`